application/controllers/UploaderController.php

```diff
@@ -237,11 +237,43 @@ public function run($actionID)
                 echo ls_json_encode($return);
                 Yii::app()->end();
             }
-
+            // event to allow updating some part
+            $event = new PluginEvent('beforeMovePublicFileUploaded');
+            $event->set('surveyId', $surveyid);
```

Comment:
Maybe we can include $preview as a parameter as well?

Reply:
Also  $realMimeType ?

Reply:
Not updatable for both : right ?

realMimeType can be found by plugin :)

Reply:
> Not updatable for both : right ?

Agree

> realMimeType can be found by plugin :)

Size and filename as well, but we are passing them as they are available.

The following could be interesting and are not being passed either:
- $aFieldName
- $aAttributes
- $maxfilesize
- $valid_extensions_array

Reply:
Maybe we could add as well an event beforeValidateFileUpload ?
That way we can perform custom validations or customize output messages.

Or said in another way, is that possible with the current event position?

Reply:
> Maybe we could add as well an event beforeValidateFileUpload ? That way we can perform custom validations or customize output messages.
> 
> Or said in another way, is that possible with the current event position?

Oh, i don't want to push 2 events : maybe move current one to start of process and add a system to don't do other test (with condition).

Reply:
I thought about moving the event a bit up in the process, could work. If so, maybe change the eventName to beforeProcessFileUpload?

Reply:
I would say the event should be a little bit up, before the size check.
https://github.com/LimeSurvey/LimeSurvey/blob/f97eba89064b5abc4ab5fcc7e05cced6c4cebdfb/application/controllers/UploaderController.php#L171

> But moving test with $size > $aAttributes['max_filesize'] adter the event.

Maybe we are allgined. :)

> $disableValidation = $event->get('disableValidation', false);

Then you start the roll on what validations are disbaled and which are not: Extension, size, MimeType, ...

Reply:
> I would say the event should be a little bit up, before the size check.
> 

No : because it muts be AFTER` $_FILES['uploadfile']['error']` check.

Then we need to test file size 2 times , the order

1. check `$_FILES['uploadfile']['error']` retuirn json if error
2. event
3. check `$aAttributes['max_filesize']`


:)